rllib/algorithms/tests/test_callbacks.py

```diff
@@ -102,39 +103,48 @@ def on_episode_created(
 class TestCallbacks(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        ray.init()
+        ray.init(num_cpus=12)
 
     @classmethod
     def tearDownClass(cls):
         ray.shutdown()
 
     def test_on_workers_recreated_callback(self):
+        tune.register_env("env", lambda cfg: CartPoleCrashing(cfg))
+
         config = (
             APPOConfig()
-            .environment(CartPoleCrashing)
-            .callbacks(OnWorkerCreatedCallbacks)
-            .rollouts(num_rollout_workers=2)
+            .environment("env")
+            .callbacks(OnWorkersRecreatedCallbacks)
+            .rollouts(num_rollout_workers=3)
             .fault_tolerance(recreate_failed_workers=True)
         )
 
-        for _ in framework_iterator(config, frameworks=("tf2", "torch")):
+        for _ in framework_iterator(config, frameworks=("torch", "tf2")):
             algo = config.build()
             original_worker_ids = algo.workers.healthy_worker_ids()
             for id_ in original_worker_ids:
                 self.assertTrue(algo._counters[f"worker_{id_}_recreated"] == 0)
+            self.assertTrue(algo._counters["total_num_workers_recreated"] == 0)
 
             # After building the algorithm, we should have 2 healthy (remote) workers.
-            self.assertTrue(len(original_worker_ids) == 2)
+            self.assertTrue(len(original_worker_ids) == 3)
 
             # Train a bit (and have the envs/workers crash a couple of times).
-            for _ in range(3):
-                algo.train()
+            for _ in range(5):
+                print(algo.train())
```

Comment:
No need to print?

Reply:
It's clearer to see the results when you are debugging/watching the test run, no? Leaving this in.